flow/layers/cacheable_layer.cc

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/cacheable_layer.h"
+#include "flutter/flow/raster_cache.h"
+
+namespace flutter {
+
+AutoCache::AutoCache(RasterCacheItem* raster_cache_item,
+                     PrerollContext* context,
+                     const SkMatrix& matrix)
+    : raster_cache_item_(raster_cache_item),
+      context_(context),
+      matrix_(matrix) {
+  if (context_ && context_->raster_cache) {
+    current_index_ = context_->raster_cached_entries->size();
+  }
+}
+
+AutoCache::~AutoCache() {
+  raster_cache_item_->PrerollFinalize(context_, matrix_);
+}
+
+AutoCache AutoCache::Create(LayerRasterCacheItem* raster_cache_item,
```

Comment:
Why are there 3 methods for this? Can't they just all be a single constructor/Create method? The only difference I see is that 2 of them make sure the raster_cache is not null before calling the setup method, but that should probably be done in all 3 and then they are all identical, aren't they?

Reply:
It's my not careful, sorry. Thank you, Fixed